### PR TITLE
PivotalApi

### PR DESCRIPTION
**Pull request by sperBruno**

### CODE REVIEW

File: src/main/java/com/fundacionjala/pivotalapi/Authentication.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import static com.jayway.restassured.RestAssured.baseURI;
+
+/**
+ * This class is used to authenticate into pivotal web application.
+ */
+public class Authentication {
+
+    private static Authentication instance;
+
+    private static final String X_TRACKER_TOKEN = "X-TrackerToken";
```
Comment (carledriss) [inline src/main/java/com/fundacionjala/pivotalapi/Authentication.java]: I would prefer to call it X_TRACKER_TOKEN_HEADER

File: src/main/java/com/fundacionjala/pivotalapi/BuildEndPoint.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
+package com.fundacionjala.pivotalapi;
+
+
```
Comment (carledriss) [inline src/main/java/com/fundacionjala/pivotalapi/BuildEndPoint.java]: Remove this unnecessary line

File: src/main/java/com/fundacionjala/pivotalapi/BuildEndPoint.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+/**
+ * This class is going to be used to build an EndPoint
+ */
+public class BuildEndPoint {
+
+    private static List<String> list = new ArrayList<String>();
+
+    private static Map<String, Response> container = new HashMap<String, Response>();
+
+    private static String endPoints;
```
Comment (carledriss) [inline src/main/java/com/fundacionjala/pivotalapi/BuildEndPoint.java]: Do we need this constant?

File: src/main/java/com/fundacionjala/pivotalapi/BuildEndPoint.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    /**
+     * This method replaces the correct values into the endpoint.
+     *
+     * @param endPoint is whole required endpoint.
+     * @return the correct built endpoint
+     */
+    public static String createEndPoint(String endPoint) {
+        endPoints = endPoint.replaceAll("[\\p{Ps}\\p{Pe}]", "");
```
Comment (carledriss) [inline src/main/java/com/fundacionjala/pivotalapi/BuildEndPoint.java]: Add a constant for the regex

File: src/main/java/com/fundacionjala/pivotalapi/BuildEndPoint.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        return createEndPoint(endPoint);
+    }
+
+    /**
+     * This method stores the values that are inside the Brackets in a list.
+     *
+     * @param endPoint contains the values inside the brackets.
+     */
+    public static void storeRequiredValues(String endPoint) {
+        Matcher m = Pattern.compile("\\[(.*?)\\]").matcher(endPoint);
```
Comment (carledriss) [inline src/main/java/com/fundacionjala/pivotalapi/BuildEndPoint.java]: Add a constant for the regex

File: src/main/java/com/fundacionjala/pivotalapi/RequestManager.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import org.apache.log4j.Logger;
+
+import static com.jayway.restassured.RestAssured.given;
+
+/**
+ * This class is used to perform post, put, get and delete requests.
+ */
+public class RequestManager {
+
+    private static final JsonParser JSON_PARSER = new JsonParser();
```
Comment (carledriss) [inline src/main/java/com/fundacionjala/pivotalapi/RequestManager.java]: Do we need this constant?

File: src/test/java/com/fundacionjala/pivotalapi/cucumber/stepdefinitions/ResourceStepDef.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    private static final Logger LOGGER = Logger.getLogger(ResourceStepDef.class.getName());
+
+    /**
+     * This method perform the step of post requests.
+     * @param endpoint
+     * @param values
+     */
+    @Given("^I send a post request (.*) with:$")
+    public void iSendAPostRequestProjectsWith(String endpoint, Map<String, Object> values) {
+        response = RequestManager.postRequest(buildEndPoint(endpoint), values);
```
Comment (carledriss) [inline src/test/java/com/fundacionjala/pivotalapi/cucumber/stepdefinitions/ResourceStepDef.java]: Add a static import for RequestManager

File: src/test/java/com/fundacionjala/pivotalapi/cucumber/stepdefinitions/ResourceStepDef.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+        LOGGER.info("Response to be stored is: " + response.prettyPrint());
+        setMapContainer(key, response);
+    }
+
+    /**
+     * This method will clean the environment by deleting the current project
+     */
+    @After("@cleanUp")
+    public void tearDown() {
+        final String Project1 = "Project1";
```
Comment (carledriss) [inline src/test/java/com/fundacionjala/pivotalapi/cucumber/stepdefinitions/ResourceStepDef.java]: This is hard-coded
Change the idea

File: src/test/resources/features/Stories/UpdateStory.feature [hunk 9]
```diff
@@ -0,0 +1,17 @@
+Feature: Update a Story in a Project
+
```
Comment (carledriss) [inline src/test/resources/features/Stories/UpdateStory.feature]: This feature file and DeleteStory.feature have the same precondition (Background)
Please join them.

File: src/test/resources/features/project/UpdateProject.feature [hunk 10]
```diff
@@ -0,0 +1,14 @@
+Feature: Update Project
+
```
Comment (carledriss) [inline src/test/resources/features/project/UpdateProject.feature]: This feature file and DeleteProject.feature have the same precondition (Background)
Please join them.

**Comment by carledriss** [comment]
@sperBruno 
Don't forget to update this PR

**Comment by sperBruno** [comment]
@carledriss. may I close this pull request because we worked on the same project as a team?

**Pull request closed without merging**